### PR TITLE
Prevent crash on entity-script-server due to logging interface.

### PR DESCRIPTION
**Pull request by kasenvr**
I've confirmed this fixes the issue.

However, the original issue was reported by Twa_hinkle.

To test: Ensure the entity script server does not crash for any reason on start. An easy provable test is to add the Explore beacon to your domain and ensure the entity server script runs. Confirm your domain hows up in the Explore list.

### CODE REVIEW

**Review by daleglass — APPROVED** [review]

File: libraries/script-engine/src/ScriptEngine.cpp [hunk 1]
```diff
@@ -130,6 +130,11 @@ static QScriptValue debugPrint(QScriptContext* context, QScriptEngine* engine) {
     // This message was sent by one of our script engines, let's try to see if we can find the source.
     // Note that the first entry in the backtrace should be "print" and is somewhat useless to us
     AbstractLoggerInterface* loggerInterface = AbstractLoggerInterface::get();
+    if (!loggerInterface) {
+        qCDebug(scriptengine_script, "%s", qUtf8Printable(message));
+        return QScriptValue();
+    }
+
     if (loggerInterface->showSourceDebugging()) {
```
Comment (ctrlaltdavid) [inline libraries/script-engine/src/ScriptEngine.cpp:138]:
```suggestion
    if (loggerInterface && loggerInterface->showSourceDebugging()) {
```
This should work, I think, and will `scriptEngine->print()` the message and `qCDebug()` the message with the filename.

**Review by ctrlaltdavid — CHANGES_REQUESTED** [review]

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR840/Vircadia-Alpha-PR840-c2efe34.tar_50067e33.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR840/Vircadia-Alpha-PR840-c2efe34.tar_50067e33.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR840/Vircadia-Alpha-PR840-c2efe34_50067e33.sh

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR840/Vircadia-Alpha-PR840-c2efe34_50067e33.dmg

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR840/Vircadia-Alpha-PR840-c8694e2.tar_4b2c71cf.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR840/Vircadia-Alpha-PR840-c8694e2_4b2c71cf.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR840/Vircadia-Alpha-PR840-c8694e2.tar_4b2c71cf.Z

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR840/Vircadia-Alpha-PR840-c8694e2_4b2c71cf.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR840/Vircadia-Alpha-PR840-c8694e2_4b2c71cf.exe

**Comment by kasenvr** [comment]
Twa_hinkle has confirmed that this solves the issue for him.

**Pull request merged**